From 353615b772201eb77786b25485bf20f4a50ce413 Mon Sep 17 00:00:00 2001
From: "Bruce A. Mah" <bmah@es.net>
Date: Wed, 21 Sep 2016 13:22:54 -0700
Subject: [PATCH] Fix divide-by-zero problem that can occur with a zero-length
 interval (this can happen at the end of a test under certain circumstances).
 Fixes #388, alternate solution from pull request in #389.

---
 src/iperf_api.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/iperf_api.c b/src/iperf_api.c
index 2600812..ee751a2 100755
--- a/src/iperf_api.c
+++ b/src/iperf_api.c
@@ -2593,7 +2593,12 @@ print_interval_results(struct iperf_test *test, struct iperf_stream *sp, cJSON *
     }
 
     unit_snprintf(ubuf, UNIT_LEN, (double) (irp->bytes_transferred), 'A');
-    bandwidth = (double) irp->bytes_transferred / (double) irp->interval_duration;
+    if (irp->interval_duration > 0.0) {
+	bandwidth = (double) irp->bytes_transferred / (double) irp->interval_duration;
+    }
+    else {
+	bandwidth = 0.0;
+    }
     unit_snprintf(nbuf, UNIT_LEN, bandwidth, test->settings->unit_format);
     
     st = timeval_diff(&sp->result->start_time, &irp->interval_start_time);
